File: Source/OldCode/Archive/2007 Phasor Protocols/PhasorProtocols/IDataFrameParsingState.cs
//*******************************************************************************************************
//  IDataFrameParsingState.cs
//  Copyright © 2009 - TVA, all rights reserved - Gbtc
//
//  Build Environment: C#, Visual Studio 2008
//  Primary Developer: James R Carroll
//      Office: PSO TRAN & REL, CHATTANOOGA - MR BK-C
//       Phone: 423/751-4165
//       Email: jrcarrol@tva.gov
//
//  Code Modification History:
//  -----------------------------------------------------------------------------------------------------
//  01/14/2005 - James R Carroll
//       Generated original version of source code.
//
//*******************************************************************************************************

namespace TVA.PhasorProtocols
{
    /// <summary>
    /// Represents a protocol independent interface representation of the parsing state of a <see cref="IDataFrame"/>.
    /// </summary>
    public interface IDataFrameParsingState : IChannelFrameParsingState<IDataCell>
    {
        /// <summary>
        /// Gets reference to the <see cref="IConfigurationFrame"/> associated with the <see cref="IDataFrame"/> being parsed.
        /// </summary>
        IConfigurationFrame ConfigurationFrame { get; }
    }
}

File: Source/OldCode/Archive/2006 Phasor Protocols/PhasorProtocols/IDataFrameParsingState.vb
'*******************************************************************************************************
'  IDataFrameParsingState.vb - Data frame parsing state interface
'  Copyright © 2005 - TVA, all rights reserved - Gbtc
'
'  Build Environment: VB.NET, Visual Studio 2005
'  Primary Developer: J. Ritchie Carroll, Operations Data Architecture [TVA]
'      Office: COO - TRNS/PWR ELEC SYS O, CHATTANOOGA, TN - MR 2W-C
'       Phone: 423/751-2827
'       Email: jrcarrol@tva.gov
'
'  Code Modification History:
'  -----------------------------------------------------------------------------------------------------
'  01/14/2005 - J. Ritchie Carroll
'       Initial version of source generated
'
'*******************************************************************************************************

''' <summary>This interface represents the protocol independent parsing state of any frame of data.</summary>
<CLSCompliant(False)> _
Public Interface IDataFrameParsingState

    Inherits IChannelFrameParsingState(Of IDataCell)

    ReadOnly Property ConfigurationFrame() As IConfigurationFrame

    Shadows ReadOnly Property Cells() As DataCellCollection

End Interface